src/commcare_cloud/commands/ansible/ops_tool.py

```diff
@@ -476,3 +481,58 @@ def chunked(iterable, n, fillvalue=None):
     # https://docs.python.org/3/library/itertools.html#itertools-recipes
     args = [iter(iterable)] * n
     return itertools.zip_longest(*args, fillvalue=fillvalue)
+
+
+class Snapshot(_AnsiblePlaybookAlias):
+    command = 'create-snapshot'
+    help = (
+        "This command creates a snapshot of your current environment's state.\n\n"
+        "State information is saved in the '~/.commcare-cloud/snapshots' directory. It is a good idea to run this"
+        "before making any major changes to your environment, as it allows you to have a record of your"
+        "environment's current state.\n\n"
+    )
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.env_info_dict = {}
+
+    def run(self, args, unknown_args):
+        self.env_info_dict
+        self.environment = get_environment(args.env_name)
+        self.env_info_dict["environment"] = args.env_name
+        self.log_commcare_cloud_statistics()
+        self.log_commcare_hq_statistics()
+        self.log_environment_settings_validation()
+    
+        self._create_snapshot_dir()
+
+    def _create_snapshot_dir(self):
+        env_name = self.env_info_dict["environment"]
+        snapshot_name = datetime.utcnow().strftime(DATE_FMT)
+        
+        self.snapshot_directory = os.path.expanduser(f"~/.commcare-cloud/snapshots/{env_name}/{snapshot_name}")
+        if not os.path.isdir(self.snapshot_directory):
+            os.makedirs(self.snapshot_directory)
+
+        file_path = os.path.join(self.snapshot_directory, "info.json")
+        with open(file_path, "w") as file:
+            json_str = json.dumps(self.env_info_dict)
+            file.write(json_str)
```

Comment:
nit: could use `json.dump` to dump directly to the file (and remove the `file.write` line)